File: kt/src/main/kotlin/com/github/ephemient/aoc2025/Day1.kt
package com.github.ephemient.aoc2025

import kotlin.math.abs
import kotlin.math.sign

fun day1(input: String): Pair<Int, Int> {
    var zeros = 0
    var turns = 0
    input.lines().fold(50) { pos, line ->
        val rotation = when (line.firstOrNull()) {
            'L' -> -line.drop(1).toInt()
            'R' -> line.drop(1).toInt()
            else -> return@fold pos
        }
        val pos2 = pos + rotation
        turns += abs(pos2 - rotation.sign) / 100
        if (pos != 0 && pos2 < 0) turns++
        pos2.mod(100).also { if (it == 0) zeros++ }
    }
    return zeros to zeros + turns
}

File: py/src/aoc2025/day1.py
def _day1(input: str) -> tuple[int, int]:
    pos, zeros, turns = 50, 0, 0
    for line in input.splitlines():
        rotation = (
            -int(line[1:])
            if line.startswith("L")
            else int(line[1:])
            if line.startswith("R")
            else None
        )
        if rotation is None:
            continue
        pos2 = pos + rotation
        turns += min(abs(pos2 - 1), abs(pos2 + 1)) // 100
        if pos and pos2 < 0:
            turns += 1
        if not (pos := pos2 % 100):
            zeros += 1
    return zeros, zeros + turns


parts = (_day1,)
